utils/lib/DataSource.ts

```diff
@@ -23,7 +23,39 @@ export interface DataSourceMutationResponse {
   };
 }
 
+export interface DataSourceContext {
+  coreDataSourceIds?: string[];
+}
+
 class DataSource extends Component<DataSourceConfig, string> {
+  private isCoreDataSource: boolean = false;
+
+  constructor(
+    identifier: string,
+    basePath: string = path.join(__dirname, '..', '..'),
+    context: DataSourceContext = {}
+  ) {
+    super(identifier, basePath);
```

Comment:
As you have pointed out below, `super` is going to call `getConfigFilePath()` and `getConfigContent`. Both have their own calls to helper methods like `removeBasePath` and node libraries like `path.dirname`, `glob.sync`, etc. I'm not 100% sure, but it's very likely that some of these functions might throw errors if we're giving them content we _know_ is invalid (no config files). That could cause some weird things like failing jobs or a bunch of static in our outputs.

We might want to revist how we approach this. As a stop-gap, we _could_ overwrite those top-level methods to be no-ops in this class...but that feels like a stop-gap not a proper solution.